src/ui/control/scale_control.js

```diff
@@ -52,40 +53,47 @@ class ScaleControl {
         return 'bottom-left';
     }
 
-    _onMove() {
-        updateScale(this._map, this._container, this.options);
+    _onUpdate() {
+        updateScale(this._map, this._container, this._language, this.options);
     }
 
     onAdd(map: Map): HTMLElement {
         this._map = map;
+        this._language = map.getLanguage();
         this._container = DOM.create('div', 'mapboxgl-ctrl mapboxgl-ctrl-scale', map.getContainer());
+        this._container.dir = 'auto';
 
-        this._map.on('move', this._onMove);
-        this._onMove();
+        this._map.on('move', this._onUpdate);
+        this._onUpdate();
 
         return this._container;
     }
 
     onRemove() {
         this._container.remove();
-        this._map.off('move', this._onMove);
+        this._map.off('move', this._onUpdate);
         this._map = (undefined: any);
     }
 
+    _setLanguage(language: string) {
+        this._language = language;
+        this._onUpdate();
```

Comment:
Nit: if we're calling this method directly, let's call it `_update`, otherwise it looks a little weird.